bin/check-all-gpg-keys-for-expiry.rb

```diff
@@ -0,0 +1,348 @@
+#! /usr/bin/env ruby
+#
+# check-all-gpg-keys-for-expiry
+#
+# DESCRIPTION:
+#   This script will check all keys' expiry dates and generate JSON message or output
+#   the string "CheckAllGpgKeysForExpiry OK: ..." (like a Nagios plugin). JSON message can be
+#   sent to Sensu client.
+#
+# OUTPUT:
+#   plain text
+#   Defaults: CRITICAL if a key is about to expire in 5 days
+#             WARNING if a key is about expire in 14 days
+#             GPG home directory is /root/.gnupg
+#             GPG executable binary is located under /usr/bin/gpg
+#             It checks secret keys
+#             It displays JSON message
+#
+# PLATFORMS:
+#   Linux, Mac OS X
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin, csv, open3, date, time, json
+#   GPG package
+#
+# USAGE:
+#   check-all-gpg-keys-for-expiry.rb -w 7 -c 2 -h ~/.gnupg -e /usr/local/bin/gpg
+#    --do-not-display-json-message yes --use-secret-key no
+#
+# LICENSE:
+#   Aditya Pahuja aditya.s.pahuja@gmail.com
+#   Copyright (c) 2016 Crown Copyright
+#   Released under the same terms as Crown (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'csv'
+require 'open3'
+require 'date'
+require 'time'
+require 'json'
+
+# == Description
+#
+# The Integer class provides maximum and minimum integer
+class Integer
+  # Number of bytes
+  N_BYTES = [42].pack('i').size
+
+  # Number of bits
+  N_BITS = N_BYTES * 8
+
+  # Maximum number for integer
+  MAX = 2**(N_BITS - 2) - 1
+
+  # Minimum number for integer
+  MIN = -MAX - 1
+end
+
+# == Description
+#
+# The Key class stores information about the key. It has id, expiry date,
```

Comment:
Can we move these into a shared lib so it might be used with other checks easier?

Reply:
I have moved these classes to the lib.